frontend/components/music_player/player_hooks.js

```diff
@@ -0,0 +1,261 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+const Player = (props) => {
+  const [timeCounter, setTimeCounter] = useState(0);
+  const [prevVol, setPrevVol] = useState(100);
+  const [volume, setVolume] = useState(100);
+  const [volumeHover, setVolumeHover] = useState(false);
+  const [shuffle, setShuffle] = useState(false);
+  const [repeat, setRepeat] = useState(false);
+  const [songsPlayed, setSongsPlayed] = useState([]);
+  const [duration, setDuration] = useState([]);
+
+  let intervals = [];
+
+  const audioEl = React.createRef();
+  const progressSlider = React.createRef();
+  const volumeSlider = React.createRef();
+
+  // first useEffect is to load songs into the player's queue
+  // cleans up the player by setting universal state to 'pause' when player is closed
+  useEffect(() => {
+    props.getSongs().then((songs) => {
+      props.loadQueue(Object.values(songs.songs));
+    });
+    return () => {
+      props.pauseSong();
+    };
+  }, []);
```

Comment:
You need to handle your dependencies for the useEffect. It shouldn't be empty here